### PR TITLE
Task6 parse response using android tools

### PR DESCRIPTION
**Pull request by kulebin**
- Processor package implemented;
- VenueProcessor and VenueListProcessor have been implemented;
- Venue model updated, Location model implemented;
- Getting Processor type by Url functionality implemented, UrlProcessorTypeMatcher class introduced

### CODE REVIEW

File: app/src/main/java/com/github/kulebin/myfoursquareapp/adapter/VenueListAdapter.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import android.widget.TextView;
 
 import com.github.kulebin.myfoursquareapp.R;
+import com.github.kulebin.myfoursquareapp.model.Location;
```
Comment (Mikhail-Ivanou) [inline app/src/main/java/com/github/kulebin/myfoursquareapp/adapter/VenueListAdapter.java]: do we really need this import?

File: app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/UrlProcessorTypeMatcher.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.github.kulebin.myfoursquareapp.dataSource;
```
Comment (Mikhail-Ivanou) [inline app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/UrlProcessorTypeMatcher.java]: Too complicated. Just use processor type as a param for loadData method
UriMatcher is good approach when you use ContentProvider and Content Resolver, when you use some notifiers and so on.
In our case - we khow which processor to use when we call loadData. So we do extra steps to get processorType.
And we also need to modify this class if we add new api calls, and even if just url changes!!!

We can add a result class to ProcessorType, and try to match it from IOnResultCallback<Result> type using reflection. But for now it's extra efforts.

File: app/src/main/java/com/github/kulebin/myfoursquareapp/view/VenueDisplayData.java [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
-        return location;
+    public String getAddress() {
+        return address;
     }
 
     public float getRating() {
         return rating;
     }
+
+    private String getFormattedAddress(final Location pLocation) {
```
Comment (Mikhail-Ivanou) [inline app/src/main/java/com/github/kulebin/myfoursquareapp/view/VenueDisplayData.java]: It can be a method of a model to be reusable

Comment (kulebin) [reply app/src/main/java/com/github/kulebin/myfoursquareapp/view/VenueDisplayData.java]: moved to Location model class

File: app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/UrlProcessorTypeMatcher.java (hunk 2 above)
Comment (kulebin) [reply app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/UrlProcessorTypeMatcher.java]: UriProcessorTypeMatcher class removed, ProcessorType is passed as param in loadData() method;

File: app/src/main/java/com/github/kulebin/myfoursquareapp/adapter/VenueListAdapter.java (hunk 1 above)
Comment (kulebin) [reply app/src/main/java/com/github/kulebin/myfoursquareapp/adapter/VenueListAdapter.java]: removed

File: app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/FoursquareDataSource.java [hunk 4]
```diff
@@ -12,24 +12,11 @@
 
     @Override
     public void fetchVenueList(final IOnResultCallback<List<Venue>> pOnResultCallback) {
-        mIDataLoader.loadData(Api.getVenuesTrendingUrl(), pOnResultCallback);
+        mIDataLoader.loadData(Api.getVenuesTrendingUrl(), pOnResultCallback, ProcessorType.VENUES_LIST);
```
Comment (Mikhail-Ivanou) [inline app/src/main/java/com/github/kulebin/myfoursquareapp/dataSource/FoursquareDataSource.java:15]: It's better to arrange arguments in other way.
You will often implement callback when you call loadData, so it's more readable to have simple args first, and implementations after
For example - url, processor, callback

Doesn't requare fix, just for your information

**Pull request merged**